src/queue.jl

```diff
@@ -13,11 +22,28 @@ length(s::Queue) = length(s.store)
 front(s::Queue) = front(s.store)
 back(s::Queue) = back(s.store)
 
+"""
+    enqueue!(s::Queue, x)
+
+### Args:
+* `s`: A queue into which to insert a value. 
+* `x`: The value to be inserted.
+
+`enqueue!` inserts the value `x` into the queue `s`.
+"""
 function enqueue!(s::Queue, x)
     push!(s.store, x)
     s
 end
 
+"""
+    dequeue!(s::Queue)
+
+### Args:
+* `s`: The queue from which to remove the least significant entry.
```

Comment:
"Least significant" sounds weird to me (but anyway remove the argument lists ;-).